src/components/RichTextEditor/plugins/linkPlugin/FloatingLink/CustomFloatingLink.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, ButtonSize, ButtonStyle } from '@components/Button/Button';
+import { LINK_CLASSES } from '@components/RichTextEditor/components';
+import { IconCheckMark, IconPen, IconTrashBin } from '@foundation/Icon';
+import {
+    floatingLinkActions,
+    submitFloatingLink,
+    useEditorRef,
+    useFloatingLinkSelectors,
+    useFloatingLinkUrlInput,
+} from '@udecode/plate';
+import React, { useState } from 'react';
+import { FloatingLink } from './FloatingLink';
+
+export const CustomFloatingLink = () => {
+    const isEditing = useFloatingLinkSelectors().isEditing();
+    const editor = useEditorRef();
+    const [isUrlInValid, setIsUrlInValid] = useState(false);
+    const urlHtmlProps = useFloatingLinkUrlInput({});
+
+    const input = (
+        <div data-test-id="floating-link-insert" className="tw-bg-white tw-rounded tw-shadow tw-p-7 tw-min-w-[400px]">
+            <FloatingLink.TextInput />
+            <div className="tw-pt-5">
+                <FloatingLink.UrlInput />
+            </div>
+            {isUrlInValid && <div className="tw-text-red-65 tw-mt-3">Url is invalid.</div>}
+
+            <div className="tw-mt-3">
+                <div className={'tw-pt-5 tw-flex tw-gap-x-3 tw-justify-end tw-border-t tw-border-t-black-10'}>
+                    <Button
+                        onClick={(e) => {
+                            setIsUrlInValid(false);
+                            floatingLinkActions.hide();
+                        }}
+                        size={ButtonSize.Medium}
+                        style={ButtonStyle.Secondary}
+                    >
+                        Cancel
+                    </Button>
+                    <Button
+                        onClick={(e) => {
+                            if (submitFloatingLink(editor)) {
+                                e?.preventDefault();
+
+                                setIsUrlInValid(false);
+                            } else {
+                                setIsUrlInValid(true);
```

Comment:
As soon as you have one time passed a wrong url and then click outside the link modal, it will not close anymore (except by clicking `Cancel`.